ga4gh/datamodel/__init__.py

```diff
@@ -187,3 +218,21 @@ def _scanDataFiles(self, dataDir, patterns):
             atexit.register(_cleanupHtslibsMess, indexDir)
         if numDataFiles == 0:
             raise exceptions.EmptyDirException(dataDir, patterns)
+
+    def getFileHandle(self, dataFile):
+        try:
```

Comment:
Is there a reason for not using somthing like
```python
if dataFile in self._memoTable:
     handle = self.memo_table[dataFile]
     self._cache.update(dataFile, handle)
else:
     handle = # rest of it.
return handle
```
I know there are subtle errors with concurrency that can be avoided by using the `try/except` approach instead of testing if the key is in the dictionary, but I tend to find the latter approach more readable. I don't think we're ever going to have multiple threads in this process, it's just not worth it in Python.